tests/Doctrine/Tests/DBAL/Types/DateImmutableTypeTest.php

```diff
@@ -44,6 +44,7 @@ public function testReturnsBindingType()
 
     public function testConvertsDateTimeImmutableInstanceToDatabaseValue()
     {
+        /** @var \DateTimeImmutable|\Prophecy\Prophecy\ObjectProphecy $date */
```

Comment:
Does `phpstan/phpstan-phpunit` handle prophecy mocks? Or there's a separate extension for them? In any event, it's better to have this type detection automated rather than manually entered every time.